pinot-controller/src/main/java/org/apache/pinot/controller/util/ServerSegmentMetadataReader.java

```diff
@@ -265,8 +284,12 @@ public List<ValidDocIdsMetadataInfo> getValidDocIdsMetadataFromServer(String tab
         List<ValidDocIdsMetadataInfo> validDocIdsMetadataInfoList =
             JsonUtils.stringToObject(validDocIdsMetadataList, new TypeReference<ArrayList<ValidDocIdsMetadataInfo>>() {
             });
-        for (ValidDocIdsMetadataInfo validDocIdsMetadataInfo : validDocIdsMetadataInfoList) {
-          validDocIdsMetadataInfos.put(validDocIdsMetadataInfo.getSegmentName(), validDocIdsMetadataInfo);
+        for (ValidDocIdsMetadataInfo validDocIdsMetadataInfo: validDocIdsMetadataInfoList) {
+          String segmentName = validDocIdsMetadataInfo.getSegmentName();
+          List<ValidDocIdsMetadataInfo> presentValidDocIdsMetadataInfo =
+              validDocIdsMetadataInfos.computeIfAbsent(segmentName, k -> new ArrayList<>());
+          presentValidDocIdsMetadataInfo.add(validDocIdsMetadataInfo);
+          validDocIdsMetadataInfos.put(segmentName, presentValidDocIdsMetadataInfo);
```

Comment:
L289-L292 can be inlined as `map.computeIfAbsent().add()`

Reply:
Done! I thought of using `addAll` as well but then i will have to do if(!empty) then get(0).getSegmentName.
So kept the iterative loop here.